Network/Socket.hs

```diff
@@ -52,16 +52,17 @@
 -- >               , addrSocketType = Stream
 -- >               }
 -- >         head <$> getAddrInfo (Just hints) mhost (Just port)
--- >     open addr = do
--- >         sock <- socket (addrFamily addr) (addrSocketType addr) (addrProtocol addr)
--- >         setSocketOption sock ReuseAddr 1
--- >         withFdSocket sock setCloseOnExecIfNeeded
--- >         bind sock $ addrAddress addr
--- >         listen sock 1024
--- >         return sock
--- >     loop sock = forever $ do
--- >         (conn, _peer) <- accept sock
--- >         void $ forkFinally (server conn) (const $ gracefulClose conn 5000)
+-- >     open addr = E.bracketOnError
+-- >         (socket (addrFamily addr) (addrSocketType addr) (addrProtocol addr))
```

Comment:
I would like to extract this part as a function like `openSocket` for readability.

```
open addr = E.bracketOnError openSocket close $ \sock -> do
```

is much easier to read for me.